write.go

```diff
@@ -286,3 +457,31 @@ func msgSize(key, value []byte) int32 {
 		sizeofBytes(key) +
 		sizeofBytes(value)
 }
+
+// Messages with magic >2 are called records. This method writes messages using message format 2.
+func writeRecord(w *bufio.Writer, attributes int8, baseTime time.Time, baseOffset int64, msg Message) {
+	buf := &bytes.Buffer{}
```

Comment:
This intermediary buffer is a bit worrying performance-wise, but we can follow up with an optimization to compute the size later.

Reply:
:+1: I'll send a separate PR for it.